src/sort.rs

```diff
@@ -13,7 +13,11 @@ pub fn by_meta(a: &Meta, b: &Meta, flags: Flags) -> Ordering {
             DirOrderFlag::None => by_name(a, b, flags),
             DirOrderFlag::Last => by_name_with_files_first(a, b, flags),
         },
-
+        SortFlag::Size => match flags.directory_order {
+            DirOrderFlag::First => by_size(a, b, flags),
```

Comment:
Could you also take into account the `DirOrderFlag`.

Reply:
i don't get the idea of this, i tried todo it like the `name`  sorting, but it produced, to me, unlogic results.

Reply:
IMHO it would be curious to take into account the `DirOrderFlag` into the size sort. If the change doesn't bring any major improvement we should try to stick as much as possible to the classic `ls`.

Reply:
I was thinking something like this
![Imgur](https://i.imgur.com/J57nHhR.png)

as ls has this
![Imgur](https://i.imgur.com/9pER0pZ.png)


The idea is that we will be separately sorting the dirs and files and depending on the flag we will have dirs before or after files

Reply:
sort with dirs first will look something like this
```rust
fn by_size_with_dirs_first(a: &Meta, b: &Meta, flags: Flags) -> Ordering {
    match (a.file_type, b.file_type) {
        (FileType::Directory { .. }, FileType::Directory { .. }) => by_size(a, b, flags),
        (FileType::Directory { .. }, _) => Ordering::Less,
        (_, FileType::Directory { .. }) => Ordering::Greater,
        _ => by_size(a, b, flags),
    }
}
```

Reply:
> as ls has this

I just learn something new